### PR TITLE
Latest version with todos for tests

### PR DESCRIPTION
**Pull request by simona1**
Will add the tests soon  and update the PR respectively - this has been addressed.

### CODE REVIEW

File: src/Actions.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import type { State } from './store';
+import fakeMessages from './messages.json';
+
+type Dispatch = ({type: string}) => void;
+type GetState = () => State;
+
+function fakePromise(data, delay) {
```
Comment (HamidAghdaee) [inline src/Actions.js]: what's the use for fake data? I thought you were connected to a live data source. please remove all references to a fake source of data.

File: src/store.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
+// @flow
```
Comment (HamidAghdaee) [inline src/store.js:1]: please create a reducers folder and put the reducers in there. you will also need to export the initial reducer from there as well

File: src/Toolbar.jsx [hunk 3]
```diff
@@ -1,33 +1,78 @@
...
+import frustrated from './images/emojis/frustrated.jpg';
+import sad from './images/emojis/sad.jpg';
+import neutral from './images/emojis/neutral.jpg';
+import happy from './images/emojis/happy.jpg';
+import smile from './images/emojis/smile.jpg';
+import convertScoreToColorAndEmoji from './convertScoreToColorAndEmoji';
+
 import './index.css';
+import store from './store';
+import Actions from './Actions';
```
Comment (HamidAghdaee) [inline src/Toolbar.jsx]: actions should be in their own folder

File: src/Toolbar.jsx [hunk 4]
```diff
@@ -1,33 +1,78 @@
...
+  };
+
+
+export default class Toolbar extends React.Component {
+  componentWillMount() {
+    const channels = Object.keys(store.getState().channelData);
+    if (channels.length === 0) {
+      store.dispatch(Actions.fetchChannels());
+    }
+    store.subscribe(() => this.forceUpdate());
```
Comment (HamidAghdaee) [inline src/Toolbar.jsx]: all connections to the store need to be done through react-redux's connect function

File: src/Toolbar.jsx [hunk 5]
```diff
@@ -1,33 +1,78 @@
...
+    smile,
+    happy,
+  };
+
+
+export default class Toolbar extends React.Component {
+  componentWillMount() {
+    const channels = Object.keys(store.getState().channelData);
+    if (channels.length === 0) {
+      store.dispatch(Actions.fetchChannels());
```
Comment (HamidAghdaee) [inline src/Toolbar.jsx]: all dispatches to the store need to be done through react-redux's connect function. The only place you should ever need to be importing the store is in src/index.js. nowhere else

File: src/index.js [hunk 6]
```diff
@@ -10,3 +14,15 @@ ReactDOM.render(
   document.getElementById('root'),
 );
 registerServiceWorker();
+
```
Comment (HamidAghdaee) [inline src/index.js:22]: The only time you're ever going to import store in your code should be in index.js and only to provide it in the provider component.

File: src/Actions.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
+// @flow
```
Comment (HamidAghdaee) [inline src/Actions.js]: don't use a default export for all actions. just individual exports for each function

**Comment by simona1** [comment]
Code has been refactored to address comments. Passes lint and flow checks and tests.

File: src/Components/MessageList.jsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    selectedChannel: ?string,
+    messages: {},
+  }
+
+  render() {
+    const { selectedChannel } = this.props;
+    let { messages } = this.props;
+    if (!messages) {
+      if (selectedChannel) {
+        setTimeout(
```
Comment (HamidAghdaee) [inline src/Components/MessageList.jsx]: The render function should just be a pure function that returns React elements (or JSX). This code should go either in lifecycle methods or another utils function

File: src/FlowTypes/Types.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+export type Id = mixed;
```
Comment (HamidAghdaee) [inline src/FlowTypes/Types.js]: Please rename this to index.js and everywhere else, where the types need to be imported, use `import * from '/FlowTypes'` or something like that. The same goes for the Constants folder as well as the Utils folder

File: src/index.js (hunk 6 above)
Comment (HamidAghdaee) [reply src/index.js:22]: Please create another file for processing messages on a socket connection,

**Comment by HamidAghdaee** [comment]
nice job @simona1. Please see my comments. Also your tests fail for me. see: 
screen shot 2017-08-23 at 10 36 40 am

Your tests should not depend on the network connection at all. See: 

https://github.com/gSchool/react-storybook/blob/cart-20-redux-alternative/src/actions/actions.test.js

**Comment by simona1** [comment]
I've addressed most of the comments - and added TODOs for the portions of the code that will need to be refactored for dashboard integration as we discussed. Code passes lint and flow checks. 

Tests pass on my computer with the mock server - see screenshot. Working on refactoring them as per your comment. Thanks so much!
screen shot 2017-08-23 at 3 02 52 pm

**Comment by HamidAghdaee** [comment]
@simona1 I am still seeing the same test fails. I'm fairly certain that it is because it is making a real fetch request, which it shouldn't be doing. See the file I referenced above. You should be using thunk with extra argument and pass the API into the thunk there, this way you'll have the ability to mock the api call.

**Pull request merged**